scripts/kconfig/kconfigfunctions.py

```diff
@@ -33,66 +34,198 @@ def _dt_units_to_scale(unit):
+        'm' or 'M'  divide by 1,048,576 (1 << 20)
+        'g' or 'G'  divide by 1,073,741,824 (1 << 30)
+    """
+
+    if doc_mode or edt is None:
         return "0"
 
-    d = dt_defines[name]
-    if d.startswith(('0x', '0X')):
-        d = int(d, 16)
-    else:
-        d = int(d)
-    d >>= _dt_units_to_scale(unit)
+    dev = edt.chosen_dev(chosen)
+
+    return _dt_to_reg_size(dev, index, unit)
 
-    return str(d)
 
-def dt_hex_val(kconf, _, name, unit=None):
+def dt_path_to_reg_addr(kconf, _, path, index=0, unit=None):
     """
-    This function looks up 'name' in the DTS generated "conf" style database
-    (generated_dts_board.conf in <build_dir>/zephyr/include/generated/)
-    and if it's found it will return the value as an hex integer.  The
-    function will divide the value based on 'unit':
+    This function takes a 'path' and looks for a EDT device at that path.
+    If it finds a EDT device, it will look to see if that device has a
+    register at the give 'index' and return the address value of that reg, if
+    not we return 0.
+
+    The function will divide the value based on 'unit':
         None        No division
         'k' or 'K'  divide by 1024 (1 << 10)
         'm' or 'M'  divide by 1,048,576 (1 << 20)
         'g' or 'G'  divide by 1,073,741,824 (1 << 30)
     """
-    if doc_mode or name not in dt_defines:
-        return "0x0"
 
-    d = dt_defines[name]
-    if d.startswith(('0x', '0X')):
-        d = int(d, 16)
-    else:
-        d = int(d)
-    d >>= _dt_units_to_scale(unit)
+    if doc_mode or edt is None:
+        return "0"
+
+    try:
+       dev = edt.get_dev(path)
+    except edtlib.EDTError:
+       return "0"
+
+    return _dt_to_reg_addr(dev, index, unit)
 
-    return hex(d)
 
-def dt_str_val(kconf, _, name):
+def dt_path_to_reg_size(kconf, _, path, index=0, unit=None):
     """
-    This function looks up 'name' in the DTS generated "conf" style database
-    (generated_dts_board.conf in <build_dir>/zephyr/include/generated/)
-    and if it's found it will return the value as string.  If it's not found we
-    return an empty string.
+    This function takes a 'path' and looks for a EDT device at that path.
+    If it finds a EDT device, it will look to see if that device has a
+    register at the give 'index' and return the size value of that reg, if
+    not we return 0.
+
+    The function will divide the value based on 'unit':
+        None        No division
+        'k' or 'K'  divide by 1024 (1 << 10)
+        'm' or 'M'  divide by 1,048,576 (1 << 20)
+        'g' or 'G'  divide by 1,073,741,824 (1 << 30)
     """
-    if doc_mode or name not in dt_defines:
-        return ""
 
-    return dt_defines[name].strip('"')
+    if doc_mode or edt is None:
+        return "0"
+
+    try:
+        dev = edt.get_dev(path)
+    except edtlib.EDTError:
+       return "0"
+
+    return _dt_to_reg_size(dev, index, unit)
+
+
+def dt_path_bool_prop(kconf, _, path, prop):
```

Comment:
Is it important that it's a boolean property?

Could generalize it a bit and call it `dt_path_has_prop` otherwise.

Reply:
Yeah, would only work for stuff declared in bindings now.

Maybe there will be stricter checks against the bindings so it works out to the same thing in practice later at least.

Reply:
Even with the bindings, since things like interrupts & reg properties get handled via their various classes in EDT we wouldn't be able to have the function test for them if we just used the Property class.

Reply:
Alright, fine with me.

Think it could be called `dt_has_bool_prop()`, to get a verb in there?

Reply:
Or `dt_path_has_bool_prop()`, if you want to be consistent with the other ones.

Wonder if `node` (or nothing) would've been better than `path`. No worries if it's awkward to change though.

Reply:
what do you mean by `node`?  Just changing the var name or something else?

Reply:
Meant that `dt_node_has_bool_prop` might be better than `dt_path_has_bool_prop`. Not a clear win though...

A `has` would be nice though, so it's not just a noun for a function that tests if the node has a property. Would confuse me.

Reply:
How do you feel about exposing or using `EDT._dt`?  I would rather this be `dt_path_has_prop`.

Reply:
Is it useful to test for non-bool properties as well?

Maybe we could come up with something later if we run into cases where we need to check if other properties exist.